packages/mobile/src/language/Language.tsx

```diff
@@ -12,20 +12,23 @@ import { CustomEventNames } from 'src/analytics/constants'
 import { setLanguage } from 'src/app/actions'
 import { AVAILABLE_LANGUAGES } from 'src/config'
 import { Namespaces } from 'src/i18n'
+import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 
 type Props = StackScreenProps<StackParamList, Screens.Language>
 
-export default function Language({ navigation, route }: Props) {
+export default function Language({ route }: Props) {
   const dispatch = useDispatch()
   const { t, i18n } = useTranslation(Namespaces.accountScreen10)
 
   function onSelect(language: string, code: string) {
     CeloAnalytics.track(CustomEventNames.language_select, { language, selectedAnswer: code })
-    const nextScreen = route.params?.nextScreen ?? Screens.JoinCelo
     dispatch(setLanguage(code))
-    navigation.navigate(nextScreen)
+    // If came from Account screen, should navigate back using back button
+    if (!route.params?.fromAccountScreen) {
```

Comment:
I do like that nextScreen is more generic here - you could flip them so the onboarding version has a nextScreen but the account screen doesn't?

Reply:
I couldn't figure out exactly why, but navigating to the the `Account` screen through the previous design was causing the `onPress` method of the `DrawerTopBar` to break. Given there are only two reachable destinations from this screen, I opted to go for a less extensible quick fix.

If you feel like it's worth figuring out right now, I can dive a little deeper.

Reply:
So in onSelect we call navigate to nextScreen - in this case when you come from account, we want you to navigate back because the language screen gets pushed onto the stack. When you're finished, we should push it off the stack from account, but during onboarding we want to push a new screen onto the stack. So I think the way I would model this is something like:`nextScreen: Screens | 'GO_BACK'`. That way you can either know to push a new screen onto the stack, or pop the current one off.